components/Common/Badge/index.tsx

```diff
@@ -0,0 +1,55 @@
+import ArrowRightIcon from '@heroicons/react/24/solid/ArrowRightIcon';
+import Link from 'next/link';
+import type { ComponentProps, FC, PropsWithChildren } from 'react';
+import styles from './index.module.scss';
+import { component } from '@/types/component';
+import classNames from 'classnames';
+
+type BadgeProps = {
+  kind?: component;
+  badgeText?: string;
+} & ComponentProps<typeof Link>;
+
+const Badge: FC<PropsWithChildren<BadgeProps>> = ({
+  kind,
+  badgeText,
+  children,
+  ...args
+}) => {
+  if (typeof kind === 'undefined') kind = component.Default;
+  const iconClasses = classNames(styles.icon, {
```

Comment:
I'm 100% against the way you defined classnames here, this is completely suboptimal and you created dozens of line of code when honestly speaking only one is needed

Usage of classNames is not even needed here and you could have a simple map of kind to className such as how @AugustinMauroy did on his PR